### ISSUE TITLE
Python: handle circular test dependencies

### ISSUE THREAD

**Issue by FRidh** [NixOS/nixpkgs#63168]
## Issue description

It's not uncommon for test tools to depend on each other in each others `checkInputs`. 

## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.

**Comment by jonringer** [comment]
@FRidh bump

I've seen the `otherpackage.overrideAttrs(oldAttrs: { doCheck = false; });` hack, but is there another solution to this?

**Comment by FRidh** [comment]
Nope. Nix creates a directed acyclic graph so it is fundamentally impossible to solve this correctly aside from breaking these loops in such a way. Ideally we would have an automated way to break the loop and then rebuild the package with tests again.

**Comment by jonringer** [comment]
@FRidh unless we have a way to recursively disable tests for an entire dependency tree, I don't see this issue going anywhere. The current paradigm of disabling testing on the few packages when it's applicable (hypothesis, pytest, very few others) is okay with me for the time being. If this does become a broader issue, then it might be worth revisiting.
